src/core/grid_scan_matcher.h

```diff
@@ -8,45 +14,99 @@
 #include "state_data.h"
 #include "sensor_data.h"
 
+/**
+ * \brief Interface of observer on Scan Matcher
+ */
 class GridScanMatcherObserver {
 public:
+  /**
+   * Virtual function. Starts matching.
+   * \param RobotState Pose of a robot
+   * \param TransformedLaserScan Information about a scan
+   * \param GridMap Information about a map
+   */
   virtual void on_matching_start(const RobotState &,           /*pose*/
                                  const TransformedLaserScan &, /*scan*/
                                  const GridMap &) {}    /*map*/
+  /**
+   * Virtual function. Tests the scan
+   * \param RobotState Pose of a robot
+   * \param TransformedLaserScan Information about a scan
+   * \param double Score
+   */
   virtual void on_scan_test(const RobotState &,           /*pose*/
                             const TransformedLaserScan &, /*scan*/
                             double) {};                  /*score*/
+  /**
+   * Virtual function. Updates a pose of a robot
+   * \param RobotState Pose of a robot
+   * \param TransformedLaserScan Information about a scan
+   * \param double Score
+   */
   virtual void on_pose_update(const RobotState &,            /*pose*/
                               const TransformedLaserScan &,  /*scan*/
                               double) {};                    /*score*/
-  virtual void on_matching_end(const RobotState &, /*delta*/
+  /*
+   * Virtual function. Stops matching
+   * \param RobotState Pose of a robot
+   * \param double Score
+   */
+	 virtual void on_matching_end(const RobotState &, /*delta*/
                                double) {};         /*best_score*/
 };
 
+/**
+ * \brief Interface of Estimator of Scan Cost.
+ * Cost - is a number that complies to a scan. As greater cost than worser scan
+ */
 class ScanCostEstimator {
 public:
+
+  /**
+   * Estimates the cost of a scan
+   * \param pose Pose of a robot
+   * \param scan Information about a scan
+   * \param map - Information about a map
+   */
   virtual double estimate_scan_cost(const RobotState &pose,
                                     const TransformedLaserScan &scan,
                                     const GridMap &map,
                                     double min_cost) = 0;
 };
 
+/**
+ * \brief  Class that matches scans
+ * When a new scan appears, it is requied to compare it with a current status of World. This class is to handle these situations
```

Comment:
"appears" -> "is available"
It isn't required to compare a scan with the world.
Consider: "Performes scan adjustment by altering robot pose in order to maximize correspondence between the scan and a grid map. The rule of correspondence computation is defined in ScanCostEstimator subclasses"